libs/sdk-core/src/models.rs

```diff
@@ -1584,6 +1584,127 @@ impl PaymentPathEdge {
     }
 }
 
+pub(crate) mod sanitize {
+    use anyhow::Result;
+    use serde::Serialize;
+
+    use crate::{
+        FullReverseSwapInfo, OpeningFeeParams, ReverseSwapInfoCached, SwapInfo, SwapStatus,
+    };
+
+    pub(crate) trait Sanitize<R, S> {
+        /// Sanitizes a raw struct [R] to a sanitized version of it [S]
+        fn sanitize(self) -> S;
+    }
+
+    pub(crate) fn sanitize_vec<R, S>(vals: Vec<R>) -> Vec<S>
+    where
+        R: Sanitize<R, S>,
+    {
+        vals.into_iter()
+            .map(|val| val.sanitize())
+            .collect::<Vec<S>>()
+    }
+
+    pub(crate) fn sanitize_vec_pretty_print<R, S>(vals: Vec<R>) -> Result<String>
+    where
+        R: Sanitize<R, S>,
+        S: Serialize,
+    {
+        serde_json::to_string_pretty(&sanitize_vec(vals)).map_err(anyhow::Error::new)
+    }
+
+    /// Sanitized version of [FullReverseSwapInfo], without the following fields:
+    /// - `preimage`
+    /// - `private_key`
+    #[derive(Serialize)]
+    pub(crate) struct FullReverseSwapInfoSanitized {
```

Comment:
You think it would be possible to do this conversion with a macro?

Have something like this on `FullReverseSwapInfo`:

```
#[secret]
pub preimage: Vec<u8>
```

And have the serializer ignore those 'secret' fields, or put `"redacted"` instead? 
That would help so we don't have to remember to update these structs when fields are added to them, and it would be easy to add this when other secrets are identified.

Reply:
If we are going this path then perhaps omitting from the json using `#[serde(skip_serializing)]` I don't think we are serializing to json anywhere else.

Reply:
> If we are going this path then perhaps omitting from the json using `#[serde(skip_serializing)]` I don't think we are serializing to json anywhere else.

That would be fine too.

Reply:
That's a good idea, but unfortunately we do serialize to json in `list_payments` > `LnPaymentDetails`, which has

```rust
    /// Only set for [PaymentType::Received] payments that were received in the context of a swap
    pub swap_info: Option<SwapInfo>,

    /// Only set for [PaymentType::Sent] payments that were sent in the context of a reverse swap
    pub reverse_swap_info: Option<ReverseSwapInfo>,
```

If we `#[serde(skip_serializing)]` any field of `SwapInfo`, serialization fails here.

Reply:
@ok300 I don't understand what you mean, can you elaborate? Where does serialization fail?

Reply:
It fails when trying to build the `LnPaymentDetails` as part of reconstructing the `Payment`:

https://github.com/breez/breez-sdk-greenlight/blob/a5657e49b6fa9bd9fa2da4dbdeff5af30cb20973/libs/sdk-core/src/persist/transactions.rs#L240-L250

The query has a `JOIN` on the swap table. From this data, it tries to reconstruct the `SwapInfo` present in the `Payment`'s `LnPaymentDetails`, which fails if some fields are marked as `#[serde(skip_serializing)`.